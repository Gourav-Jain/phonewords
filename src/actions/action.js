/**
 * Action Types
 */
export const PHONE_WORD_REQS = "PHONE_WORD_REQS";
export const PHONE_WORD_RESP = "PHONE_WORD_RESP";
export const PHONE_WORD_FAIL = "PHONE_WORD_FAIL";

/**
 * Action Creator
 */
export const setPhonewordRequest = (payload) => ({
  type: PHONE_WORD_REQS,
  payload,
});
export const setPhonewordResponse = (payload) => ({
  type: PHONE_WORD_RESP,
  payload,
});
export const setPhonewordFailer = (payload) => ({
  type: PHONE_WORD_FAIL,
  payload,
});
