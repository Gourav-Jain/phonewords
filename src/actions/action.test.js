import * as actions from "./action";

describe("@actions and @creators", () => {
  describe("actions", () => {
    const phonewordActions = [
      actions.PHONE_WORDS_REQS,
      actions.PHONE_WORDS_RESP,
      actions.PHONE_WORDS_FAIL,
    ];

    test.each(phonewordActions)("should return %s action", (actionType) =>
      expect(actions[actionType]).toMatchSnapshot()
    );
  });

  describe("creators", () => {
    test.each`
      creatorName               | mockData
      ${"setPhonewordRequest"}  | ${"mock-data"}
      ${"setPhonewordResponse"} | ${"mock-data"}
      ${"setPhonewordFailer"}   | ${"mock-data"}
    `("$creatorName returns correctly", ({ creatorName, mockData }) => {
      expect(actions[creatorName](mockData)).toMatchSnapshot();
    });
  });
});
