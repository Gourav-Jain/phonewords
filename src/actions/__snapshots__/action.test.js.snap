// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@actions and @creators actions should return undefined action 1`] = `undefined`;

exports[`@actions and @creators actions should return undefined action 2`] = `undefined`;

exports[`@actions and @creators actions should return undefined action 3`] = `undefined`;

exports[`@actions and @creators creators setPhonewordFailer returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "PHONE_WORD_FAIL",
}
`;

exports[`@actions and @creators creators setPhonewordRequest returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "PHONE_WORD_REQS",
}
`;

exports[`@actions and @creators creators setPhonewordResponse returns correctly 1`] = `
Object {
  "payload": "mock-data",
  "type": "PHONE_WORD_RESP",
}
`;
